src/config.rs

```diff
@@ -204,6 +211,40 @@ impl Config {
         self.search = Some(search);
         self.last_search = LastSearch::Search;
     }
+
+    /// Normalize config, limit nameservers max 3, domain max 6.
+    pub fn glibc_normalize(&mut self) {
+        self.nameservers.truncate(NAMESERVER_LIMIT);
+        self.search = self.search.take().map(|mut s| {
+            s.truncate(SEARCH_LIMIT); 
+            s
+        });
+    }
+
+    /// Get nameserver or on the local machine
+    pub fn get_nameservers_or_local(&self) -> Vec<ScopedIp> {
+        if self.nameservers.is_empty() {
+            vec![
+                ScopedIp::from(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1))), 
+                ScopedIp::from(IpAddr::V6(Ipv6Addr::new(0, 0, 0, 0, 0, 0, 0, 1))),
+            ]
+        } else {
+            self.nameservers.to_vec()
+        }
+    }
+
+    /// Get domain if not exists return hostname.
+    #[cfg(feature = "system")]
+    pub fn get_system_domain(&self) -> Option<String> {
+        let hn = get_hostname().map(|s| {
+            if let Some(pos) = s.find('.') {
+                return s[pos + 1..].to_string();
+            };
+            s 
```

Comment:
> the domain part is taken  to  be everything after the first '.'.  Finally, if the hostname does not contain a domain part, the root domain is assumed.

So it's either empty string here or None, but not host name. I think None is better.